### PR TITLE
Rewrite Tracing module

### PR DESCRIPTION
**Pull request by richvdh**
Rather than using a `Layer` which sends content to the JS console, construct an entire `Subscriber`, which allows us better control over what format is used, and includes data attached to spans.

### CODE REVIEW

File: src/tracing.rs [hunk 1]
```diff
@@ -1,3 +1,11 @@
+use std::{
+    fmt,
+    sync::{Arc, Mutex, Once},
+};
+
+use ::tracing::Level;
```
Comment (jplatte) [inline src/tracing.rs:6]: The leading `::` should not be needed, since there is no local symbol called `tracing` in scope.

File: src/tracing.rs [hunk 2]
```diff
@@ -31,257 +39,117 @@ pub enum LoggerLevel {
...
+}
 
-    struct Layer {
-        min_level: Level,
-        enabled: bool,
+#[wasm_bindgen]
+impl Tracing {
+    /// Check whether the `tracing` feature has been enabled.
+    #[wasm_bindgen(js_name = "isAvailable")]
+    pub fn is_available() -> bool {
```
Comment (jplatte) [inline src/tracing.rs:68]: Kept for backwards-compatibility?

File: src/tracing.rs [hunk 3]
```diff
@@ -31,257 +39,117 @@ pub enum LoggerLevel {
...
-        fn turn_on(&mut self) {
-            self.enabled = true;
-        }
-
-        fn turn_off(&mut self) {
-            self.enabled = false;
-        }
+    fn install_or_get_inner() -> Arc<Mutex<TracingInner>> {
+        static mut INSTALL: Option<Arc<Mutex<TracingInner>>> = None;
+        static INSTALLED: Once = Once::new();
```
Comment (jplatte) [inline src/tracing.rs:74]: `OnceLock` is stable, can you not use that instead of a separate `Once` and `static mut`?

File: src/tracing.rs (hunk 1 above)
Comment (richvdh) [reply src/tracing.rs:6]: hrm, my IDE seemed to want to use the current file. Can try removing it.

File: src/tracing.rs (hunk 2 above)
Comment (richvdh) [reply src/tracing.rs:68]: yes, basically.

File: src/tracing.rs (hunk 3 above)
Comment (richvdh) [reply src/tracing.rs:74]: probably, this bit was cargo-culted from the previous impl! Will take a look.

**Review by jplatte — APPROVED** [review]

**Pull request merged**